packages/cli/src/commands/validator/status.ts

```diff
@@ -0,0 +1,113 @@
+import { Address } from '@celo/contractkit'
+import { eqAddress } from '@celo/utils/lib/address'
+import { concurrentMap } from '@celo/utils/lib/async'
+import { bitIsSet, parseBlockExtraData } from '@celo/utils/lib/istanbul'
+import { flags } from '@oclif/command'
+import { cli } from 'cli-ux'
+import { BaseCommand } from '../../base'
+import { newCheckBuilder } from '../../utils/checks'
+import { Flags } from '../../utils/command'
+
+export default class ValidatorStatus extends BaseCommand {
+  static description =
+    'Show information about whether the validator signer is elected and validating. This command will check that the validator meets the registration requirements, and its signer is currently elected and actively signing blocks.'
+
+  static flags = {
+    ...BaseCommand.flags,
+    signer: Flags.address({
+      description: 'address of the validator to check if elected and validating',
+    }),
+    validator: Flags.address({
+      description: 'address of the signer to check if elected and validating',
+    }),
+    lookback: flags.integer({
+      description: 'how many blocks to look back for signer activity',
+      default: 100,
+    }),
+  }
+
+  static examples = [
+    'status --validator 0x5409ED021D9299bf6814279A6A1411A7e866A631',
+    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c',
+    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c --lookback 100',
+  ]
+
+  requireSynced = true
+
+  async run() {
+    const res = this.parse(ValidatorStatus)
+
+    // Check that the specified validator or signer meets the validator requirements.
+    const checker = newCheckBuilder(this, res.flags.signer)
+    if (res.flags.validator) {
+      const account = res.flags.validator
+      checker.isValidator(account).meetsValidatorBalanceRequirements(account)
+    } else if (res.flags.signer) {
+      checker.signerMeetsValidatorBalanceRequirements().signerAccountIsValidator()
```

Comment:
Rather than add additional checks, why don't we just pull the `account` from the signer and run the same checks on the account?

Reply:
At this point in the code it is unclear whether the accounts given are valid and I wanted to avoid an opaque call reversion. I realized that as coded it still has that issue, so I've added an `isAccount` check.